### PR TITLE
Add Debounce and Retries to respawn screen

### PR DESCRIPTION
**Pull request by mattyoung-improbable**
#### Description
Use a coroutine to retry respawn command, also prevent multiple commands being send by mashing the spacebar

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs [hunk 1]
```diff
@@ -32,6 +33,7 @@ private struct CameraSettings
         private GroundChecker groundChecker;
         private FpsAnimator fpsAnimator;
         private GunManager currentGun;
+        private SpatialOSComponent spatial;
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs]: Not a fan of this variable name

File: workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs [hunk 2]
```diff
@@ -61,6 +66,8 @@ private void OnEnable()
             Cursor.lockState = CursorLockMode.Locked;
             Cursor.visible = false;
             serverMovement.OnForcedRotation += OnForcedRotation;
+            health.OnRespawn += OnRespawn;
+            spatial = GetComponent<SpatialOSComponent>();
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs]: Move this up to group with the other `GetComponent`

File: workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs (hunk 1 above)
Comment (mattyoung-improbable) [reply workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs]: done

File: workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs (hunk 2 above)
Comment (mattyoung-improbable) [reply workers/unity/Assets/Fps/Scripts/GameLogic/FpsDriver.cs]: done

**Pull request merged**